agent.go

```diff
@@ -297,6 +306,11 @@ func (a *Agent) pingCandidate(local, remote *Candidate) {
 	}
 
 	a.log.Tracef("ping STUN from %s to %s\n", local.String(), remote.String())
+	if overflow := len(a.pendingBindingRequests) - maxPendingBindingRequests; overflow > 1 {
+		a.pendingBindingRequests = a.pendingBindingRequests[overflow:]
```

Comment:
This LRU implementation is clever. I had to think about it for a minute before I understood it. My only concern here is that it might be worth logging a message (at DEBUG or higher) indicating that N unanswered binding requests are being culled. If there is a big flurry of bind requests generated (think multiple interfaces with ipv4/ipv6 addresses) I could see a situation with a peer responding to some requests slower than others and then the resulting rejections might be confusing without this additional piece of information.

Reply:
ACK! Will add that before merging